api/scpca_portal/models/sample.py

```diff
@@ -17,109 +20,164 @@ class Meta:
-    project.has_cite_seq_data = has_cite_seq_data
-    project.has_spatial_data = has_spatial_data
-    project.save()
-
-    for summary, count in summaries.items():
-        try:
-            project_summary = ProjectSummary.objects.get(
-                project=project, diagnosis=summary[0], seq_unit=summary[1], technology=summary[2]
+    def __str__(self):
+        return f"Sample {self.scpca_id} of {self.project}"
+
+    @classmethod
+    def create_from_dict(cls, data, project):
+        # First figure out what metadata is additional. This varies project by
+        # project, so whatever's not on the Sample model is additional.
+        sample_columns = (
+            "age",
+            "cell_count",
+            "diagnosis",
+            "disease_timing",
+            "scpca_library_id",  # Also include this because we don't want it in additional_metadata.
+            "scpca_sample_id",
+            "seq_units",
+            "sex",
+            "subdiagnosis",
+            "technologies",
+            "tissue_location",
+            "treatment",
+        )
+        additional_metadata = {k: v for k, v in data.items() if k not in sample_columns}
+
+        sample = Sample(
+            additional_metadata=additional_metadata,
+            age_at_diagnosis=data["age"],
+            cell_count=data["cell_count"],
+            diagnosis=data["diagnosis"],
+            disease_timing=data["disease_timing"],
+            project=project,
+            scpca_id=data["scpca_sample_id"],
+            seq_units=data.get("seq_units", ""),
+            sex=data["sex"],
+            subdiagnosis=data["subdiagnosis"],
+            technologies=data.get("technologies", ""),
+            tissue_location=data["tissue_location"],
+            treatment=data.get("treatment"),
+        )
+        sample.save()
+
+        return sample
+
+    @staticmethod
+    def get_output_metadata_path(scpca_sample_id):
+        return os.path.join(common.OUTPUT_DATA_DIR, f"{scpca_sample_id}_libraries_metadata.tsv")
+
+    @staticmethod
+    def get_output_spatial_metadata_path(scpca_sample_id):
+        return os.path.join(common.OUTPUT_DATA_DIR, f"{scpca_sample_id}_spatial_metadata.tsv")
+
+    # TODO(arkid15r): remove the property after BE/FE refactoring.
+    @property
+    def computed_file(self):
+        return self.sample_computed_file.first()
+
+    @property
+    def output_single_cell_data_file_name(self):
+        return f"{self.scpca_id}.zip"
+
+    @property
+    def output_single_cell_data_file_path(self):
+        return os.path.join(common.OUTPUT_DATA_DIR, self.output_single_cell_data_file_name)
+
+    @property
+    def output_spatial_data_file_name(self):
+        return f"{self.scpca_id}_spatial.zip"
+
+    @property
+    def output_spatial_data_file_path(self):
+        return os.path.join(common.OUTPUT_DATA_DIR, self.output_spatial_data_file_name)
+
+    def create_single_cell_data_file(self, libraries_metadata: List[Dict], workflow_version: str):
```

Comment:
do you think this makes more sense in the ComputedFile instead of Sample? same with project?